sktime/regression/base.py

```diff
@@ -151,32 +157,43 @@ def fit(self, X, y):
         ending in "_" and sets is_fitted flag to True.
         """
-        start = int(round(time.time() * 1000))
-        # convenience conversions to allow user flexibility:
-        # if X is 2D array, convert to 3D, if y is Series, convert to numpy
-        X, y = _internal_convert(X, y)
-        X_metadata = _check_regressor_input(
-            X, y, return_metadata=self.METADATA_REQ_IN_CHECKS
-        )
-        self._X_metadata = X_metadata
-        missing = X_metadata["has_nans"]
-        multivariate = not X_metadata["is_univariate"]
-        unequal = not X_metadata["is_equal_length"]
-        # Check this regressor can handle characteristics
-        self._check_capabilities(missing, multivariate, unequal)
-        # Convert data as dictated by the regressor tags
-        X = self._convert_X(X)
-        multithread = self.get_tag("capability:multithreading")
-        if multithread:
-            try:
-                self._threads_to_use = check_n_jobs(self.n_jobs)
-            except NameError:
-                raise AttributeError(
-                    "self.n_jobs must be set if capability:multithreading is True"
-                )
+        y = self._check_y(y)
+
+        self._is_vectorized = isinstance(y, VectorizedDF)
+        # we call the ordinary _fit if no looping/vectorization needed
+        if not self._is_vectorized or self.get_tag("capability:multioutput"):
+            # reset estimator at the start of fit
+            self.reset()
+            start = int(round(time.time() * 1000))
+            # convenience conversions to allow user flexibility:
+            # if X is 2D array, convert to 3D, if y is Series, convert to numpy
+            X, y = _internal_convert(X, y)
+            X_metadata = _check_regressor_input(
+                X, y, return_metadata=self.METADATA_REQ_IN_CHECKS
+            )
+            self._X_metadata = X_metadata
+            missing = X_metadata["has_nans"]
+            multivariate = not X_metadata["is_univariate"]
+            unequal = not X_metadata["is_equal_length"]
+            # Check this regressor can handle characteristics
+            self._check_capabilities(missing, multivariate, unequal)
+            # Convert data as dictated by the regressor tags
+            X = self._convert_X(X)
+            multithread = self.get_tag("capability:multithreading")
+            if multithread:
+                try:
+                    self._threads_to_use = check_n_jobs(self.n_jobs)
+                except NameError:
+                    raise AttributeError(
+                        "self.n_jobs must be set if capability:multithreading is True"
+                    )
+
+            self._fit(X, y)
+            self.fit_time_ = int(round(time.time() * 1000)) - start
+        else:
+            # otherwise we call the vectorized version of fit
+            self._vectorize("fit", X=X, y=y)
 
-        self._fit(X, y)
-        self.fit_time_ = int(round(time.time() * 1000)) - start
```

Comment:
I would keep the `fit_time_` outside the loop.

Reply:
That makes sense. I think `start` should also be outside.